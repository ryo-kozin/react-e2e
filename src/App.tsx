import React, { useState } from 'react';

const App: React.FC = () => {
	const [count, setCount] = useState(0);

	return (
		<div>
			<h1>Counter</h1>
			<p data-testid='count'>{count}</p>
			<button onClick={() => setCount(count + 1)}>Increment</button>
			<button onClick={() => setCount(count - 1)}>Decrement</button>
		</div>
	);
};

export default App;
