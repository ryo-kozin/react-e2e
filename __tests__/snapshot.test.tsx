import React from 'react';
import '@testing-library/jest-dom';
import App from '../src/App';
import renderer from 'react-test-renderer';

test('matches snapshot', () => {
	const tree = renderer.create(<App />).toJSON();
	expect(tree).toMatchSnapshot();
});
