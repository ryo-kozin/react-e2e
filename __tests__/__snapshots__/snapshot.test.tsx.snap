// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <h1>
    Counter
  </h1>
  <p
    data-testid="count"
  >
    0
  </p>
  <button
    onClick={[Function]}
  >
    Increment
  </button>
  <button
    onClick={[Function]}
  >
    Decrement
  </button>
</div>
`;
